File: MediCareWebApi/DTOs/Request/AppointmentRequestDTO.cs
using System.ComponentModel.DataAnnotations;
using MediCare.Enums;

namespace MediCare.DTOs.Request
{
    public class AppointmentRequestDTO
    {
        public int? Id { get; set; }

        [Required(ErrorMessage = "Data wizyty jest wymagana.")]
        [DataType(DataType.DateTime, ErrorMessage = "Nieprawidłowy format daty i godziny.")]
        public DateTime Time { get; set; }

        public AppointmentStatus? Status { get; set; }

        public int? DoctorsUserId { get; set; }

        public int? ServiceId { get; set; }

        [MaxLength(256)]
        public string? Diagnosis { get; set; }
	}
}

File: MediCareWebSpa/src/app/DTOs/request/appointment-request.dto.ts
import { AppointmentStatus } from "../../enums/appointment-status";

export class AppointmentRequestDTO {
  id: number;
  time: Date;
  status: AppointmentStatus;
  diagnosis: string;
  doctorsUserId?: number;
  serviceId?: number;

  constructor(
    id: number,
    time: Date,
    status: AppointmentStatus,
    diagnosis: string,
    doctorsUserId?: number,
    serviceId?: number
  ) {
    this.id = id;
    this.time = time;
    this.status = status;
    this.diagnosis = diagnosis;
    this.doctorsUserId = doctorsUserId;
    this.serviceId = serviceId;
  }
}
